src/foam/nanos/cron/CronScheduler.java

```diff
@@ -36,31 +38,33 @@ private Date getMinScheduledTime() {
     return ((Cron) min.getValue()).getScheduledTime();
   }
 
-  private final Thread cronJobs = new Thread() {
-    @Override
-    public void run() {
-      try {
-        while (true) {
-          Date dtnow = new Date();
-          cronDAO_.where(MLang.LTE(Cron.SCHEDULED_TIME, dtnow)).select(new AbstractSink() {
-            @Override
-            public void put(FObject obj, Detachable sub) {
-              ((Cron) obj).runScript(getX());
-              cronDAO_.put(obj);
-            }
-          });
+  public void start() {
+    cronDAO_ = (DAO) getX().get("cronDAO");
+    new Thread(this).start();
+  }
+
+  @Override
+  public void run() {
+    NanoLogger logger = (NanoLogger) getX().get("logger");
+    final PM pm = new PM(this.getClass(), "cronScheduler");
 
-          Date minScheduledTime = getMinScheduledTime();
-          Thread.sleep(minScheduledTime.getTime() - dtnow.getTime());
-        }
-      } catch (InterruptedException e) {
-        e.printStackTrace();
+    try {
+      while ( true ) {
+        Date now = new Date();
+        cronDAO_.where(MLang.LTE(Cron.SCHEDULED_TIME, now)).select(new AbstractSink() {
+          @Override
+          public void put(FObject obj, Detachable sub) {
+            ((Cron) obj).runScript(CronScheduler.this.getX());
+            cronDAO_.put(obj);
+            pm.log(getX());
```

Comment:
Put this in a finally block so that it PM's even if the cronjob throws an Exception.;